File: MmoGameUnity/Assets/Game/Hoftix/Script/GenerateScripts/UIModules/LoginPanelView.cs
﻿using ZJYFrameWork.UISerializable.UIInitView;
using UnityEngine;
using UnityEngine.UI;
using ZJYFrameWork.Hotfix.UISerializable;

namespace ZJYFrameWork.UISerializable
{
    public class LoginPanelView:UIViewInterface
    {
        public UnityEngine.GameObject LoginPart=null;
		public ZJYFrameWork.UISerializable.LoginPartView LoginPartView=null;
		public ZJYFrameWork.UISerializable.RegisterPartView RegisterPartView=null;
		public LoginTapToStartView LoginTapToStartView=null;
		public LoginController LoginController=null;
		public UnityEngine.GameObject tips=null;
		public UnityEngine.UI.Text UserNameText=null;
		


        public void Init(UIView _view)
        {
            LoginPart=_view.GetObjType<UnityEngine.GameObject>("LoginPart");
			LoginPartView=_view.GetObjType<ZJYFrameWork.UISerializable.LoginPartView>("LoginPartView");
			RegisterPartView=_view.GetObjType<ZJYFrameWork.UISerializable.RegisterPartView>("RegisterPartView");
			LoginTapToStartView=_view.GetObjType<LoginTapToStartView>("LoginTapToStartView");
			LoginController=_view.GetObjType<LoginController>("LoginController");
			tips=_view.GetObjType<UnityEngine.GameObject>("tips");
			UserNameText=_view.GetObjType<UnityEngine.UI.Text>("UserNameText");
			
        }
    }
}

File: DemoMMOUnity/Lua/Game/GenerateScripts/UIModules/LoginPanelView.lua
---@class LoginPanelView
LoginPanelView = class("LoginPanelView")
function LoginPanelView:Init(view)
	self._UIView = view:GetComponent("UIView")
	self.bg_Image = self._UIView:GetObjTypeStr("bg_Image") or CS.UnityEngine.UI.Image
	self.Gonggao_Button = self._UIView:GetObjTypeStr("Gonggao_Button") or CS.UnityEngine.UI.Button
	self.tips = self._UIView:GetObjTypeStr("tips") or CS.UnityEngine.GameObject
	self.UserNameText = self._UIView:GetObjTypeStr("UserNameText") or CS.UnityEngine.UI.Text
	self.LoginPart = self._UIView:GetObjTypeStr("LoginPart") or CS.UnityEngine.GameObject
	self.LoginPart_UISerializableKeyObject = self._UIView:GetObjTypeStr("LoginPart_UISerializableKeyObject") or CS.ZJYFrameWork.UISerializable.UISerializableKeyObject
	self.RegisterPart_UISerializableKeyObject = self._UIView:GetObjTypeStr("RegisterPart_UISerializableKeyObject") or CS.ZJYFrameWork.UISerializable.UISerializableKeyObject
	self.LoginStart_CanvasGroup = self._UIView:GetObjTypeStr("LoginStart_CanvasGroup") or CS.UnityEngine.CanvasGroup
	self.LoginStart_UISerializableKeyObject = self._UIView:GetObjTypeStr("LoginStart_UISerializableKeyObject") or CS.ZJYFrameWork.UISerializable.UISerializableKeyObject
	self.LoginController = self._UIView:GetObjTypeStr("LoginController") or CS.ZJYFrameWork.Hotfix.UISerializable.LoginUIController
	self.AudioSource = self._UIView:GetObjTypeStr("AudioSource") or CS.UnityEngine.AudioSource
end

